src/main/java/eu/europeana/metis/sandbox/service/record/RecordServiceImpl.java

```diff
@@ -27,32 +29,26 @@ public RecordServiceImpl(RecordRepository recordRepository, XmlRecordProcessorSe
   }
 
   @Override
-  @Transactional(isolation = Isolation.READ_UNCOMMITTED)
+  @Transactional
   public void setEuropeanaIdAndProviderId(Record recordToUpdate) {
     final String datasetId = recordToUpdate.getDatasetId();
     final String providerId = xmlRecordProcessorService.getProviderId(recordToUpdate.getContent());
     final String europeanaId = EuropeanaIdCreator.constructEuropeanaIdString(providerId, datasetId);
 
-    RecordEntity recordEntityProviderCheck = recordRepository.findByProviderIdAndDatasetId(providerId, datasetId);
-    RecordEntity recordEntityEuropeanaCheck = recordRepository.findByEuropeanaIdAndDatasetId(europeanaId, datasetId);
-
-    if (recordEntityProviderCheck == null && recordEntityEuropeanaCheck == null) {
+    final int updatedRecords = recordRepository.updateEuropeanaIdAndProviderId(recordToUpdate.getRecordId(), europeanaId, providerId, datasetId);
+    if (updatedRecords == 0) {
```

Comment:
I believe it's very unlikely, but what if the number of `updatedRecords` is > 1? Is that even possible to happen?
If it does happen, do we throw an error?